test/Common/DurableTaskEndToEndTests.cs

```diff
@@ -1483,6 +1515,54 @@ public async Task UnhandledActivityException(bool extendedSessions)
             }
         }
 
+        /// <summary>
+        /// End-to-end test which validates the handling of unhandled exceptions generated from activity code
+        /// within a sub-orchestration.
+        /// </summary>
+        [Theory]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task UnhandledSubOrchestratorActivityException(bool extendedSessions)
+        {
+            using (JobHost host = TestHelpers.GetJobHost(
+                this.loggerProvider,
+                nameof(this.UnhandledSubOrchestratorActivityException),
+                extendedSessions))
+            {
+                await host.StartAsync();
+
+                string exceptionMessage = "Kah-BOOOOM!!!";
+                var args = new StartOrchestrationArgs
+                {
+                    FunctionName = nameof(TestOrchestrations.ThrowOrchestrator),
+                    Input = exceptionMessage,
+                };
+
+                var timeout = Debugger.IsAttached ? TimeSpan.FromMinutes(5) : TimeSpan.FromSeconds(60);
+                var client = await host.StartOrchestratorAsync(
+                    nameof(TestOrchestrations.CallOrchestrator),
+                    args,
+                    this.output);
+                var status = await client.WaitForCompletionAsync(timeout, this.output);
+
+                Assert.Equal(OrchestrationRuntimeStatus.Failed, status?.RuntimeStatus);
+
+                string output = (string)status?.Output;
+                this.output.WriteLine($"Orchestration output string: {output}");
+                Assert.StartsWith(
+                    string.Format(
+                        "Orchestrator function '{0}' failed: The orchestrator function '{1}' failed: \"The activity function '{2}' failed: \"{3}\"",

```

Comment:
Out of curiosity, how far does this recursion go? Can your sub-orchestrations start their own orchestrations? If so, are orchestrations at recursion of depth 2 or greater properly handled?

Reply:
There isn't actually any recursion. In this case it's _orchestrator_ --> _subOrchestrator_ --> _activity_. It's possible for us to write a test which does recursion and/or goes deeper than 2, but I didn't think it would be necessary to cover this (I'm being optimistic that there aren't other more subtle issues that pop up as you go deeper).